ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/UtxoLedgerService.java

```diff
@@ -247,6 +248,30 @@ void sendUpdatedTransactionBuilder(
             @NotNull FlowSession session
     );
 
+    /**
+     * Sends transaction to counterparty whose session is open for.
+     *
+     * @param sessions The counterparty who are connected to {@link FlowSession} to send transaction.
+     * @param signedTransaction The {@link UtxoSignedTransaction} by signatories.
+     * @throws IllegalArgumentException if transaction fails verification before sending transaction.
+     */
+    @Suspendable
+    void sendTransaction(
+            @NotNull List<FlowSession> sessions,
+            @NotNull UtxoSignedTransaction signedTransaction
+    );
+
+    /**
+     * Receives verified transaction from counterparty whose session is open for and persists it to a vault.
+     *
+     * @param session The counterparty who are connected to {@link FlowSession} to receive transaction from.
+     * @return Returns {@link UtxoSignedTransaction} received from counterparty through session.
+     * @throws TransactionVerificationException if the transaction received fails verification.
+     */
+    @Suspendable
+    UtxoSignedTransaction receiveTransaction(
```

Comment:
can this return a `null` transaction? If not, maybe we need a `@NotNull` to the return type as well.